pandas/core/sorting.py

```diff
@@ -306,10 +306,17 @@ def nargsort(
     non_nan_idx = idx[~mask]
 
     nan_idx = np.nonzero(mask)[0]
+
     if not ascending:
         non_nans = non_nans[::-1]
         non_nan_idx = non_nan_idx[::-1]
-    indexer = non_nan_idx[non_nans.argsort(kind=kind)]
+    try:
+        indexer = non_nan_idx[non_nans.argsort(kind=kind)]
+    except TypeError:
+        # For compatibility with Series: fall back to quicksort
```

Comment:
we have a test that hits this?

Reply:
Sorry, this is a remnant from trying to bring `Series.sort_values` in line with `Index.sort_values`.  Reverted for now. This change doesn't belong in this PR. I'll bring it back in the next PR when I'll be synchronizing `Index` and `Series` behavior.